test/measurement_calibration/generate_data.py

```diff
@@ -0,0 +1,226 @@
+    """
+    create measurement calibration circuit and a GHZ state circuit for the tests
+
+    Returns:
+        QuantumCircuit: the measurement calibration circuit
+        list: the mitigation pattern
+        QuantumCircuit: ghz circuit with 5 qubits (3 are used)
+
+    """
+    meas_calibs, state_labels = complete_meas_cal(qubit_list=[1, 2, 3], qr=5)
+
+    # Choose 3 qubits
+    q_1 = 1
+    q_2 = 2
+    q_3 = 3
+    ghz = qiskit.QuantumCircuit(5, 3)
+    ghz.h(q_1)
+    ghz.cx(q_1, q_2)
+    ghz.cx(q_1, q_3)
+    ghz.measure(q_1, 0)
+    ghz.measure(q_2, 1)
+    ghz.measure(q_3, 2)
+    ghz.measure(q_3, 2)
+    return meas_calibs, state_labels, ghz
+
+
+def tensored_calib_circ_creation():
+    """
+    create tensored measurement calibration circuit and a GHZ state circuit for the tests
+
+    Returns:
+        QuantumCircuit: the tensored measurement calibration circuit
+        list: the mitigation pattern
+        QuantumCircuit: ghz circuit with 5 qubits (3 are used)
+
+    """
+    mit_pattern = [[2], [4, 1]]
+    qr = qiskit.QuantumRegister(5)
+    # Generate the calibration circuits
+    meas_calibs, mit_pattern = tensored_meas_cal(mit_pattern, qr=qr)
+
+    cr = qiskit.ClassicalRegister(3)
+    ghz_circ = qiskit.QuantumCircuit(qr, cr)
+    ghz_circ.h(qr[2])
+    ghz_circ.cx(qr[2], qr[4])
+    ghz_circ.cx(qr[2], qr[1])
+    ghz_circ.measure(qr[2], cr[0])
+    ghz_circ.measure(qr[4], cr[1])
+    ghz_circ.measure(qr[1], cr[2])
+    return meas_calibs, mit_pattern, ghz_circ
+
+
+def meas_calibration_circ_execution(shots: int, seed: int):
+    """
+    create measurement calibration circuits and simulates them with noise
+    Args:
+        shots (int): number of shots per simulation
+        seed (int): the seed to use in the simulations
+
+    Returns:
+        list: list of Results of the measurement calibration simulations
+        list: list of all the possible states with this amount of qubits
+        dict: dictionary of results counts of bell circuit simulation with measurement errors
+    """
+    # define the circuits
+    meas_calibs, state_labels, ghz = meas_calib_circ_creation()
+
+    # define noise
+    prob = 0.2
+    error_meas = pauli_error([('X', prob), ('I', 1 - prob)])
+    noise_model = NoiseModel()
+    noise_model.add_all_qubit_quantum_error(error_meas, "measure")
+
+    # run the circuits multiple times
+    backend = qiskit.Aer.get_backend('qasm_simulator')
+    cal_results = qiskit.execute(meas_calibs, backend=backend, shots=shots, noise_model=noise_model,
+                                 seed_simulator=seed).result()
+
+    bell_results = qiskit.execute(ghz, backend=backend, shots=shots, noise_model=noise_model,
+                                  seed_simulator=seed).result().get_counts()
+
+    return cal_results, state_labels, bell_results
+
+
+def tensored_calib_circ_execution(shots: int, seed: int):
+    """
+    create tensored measurement calibration circuits and simulates them with noise
```

Comment:
ther "create ... simulate" or "creates ... simulates". The first option is more preferable.